fulltext/backends/__gz.py

```diff
@@ -0,0 +1,32 @@
+import gzip
+from os.path import splitext, basename
+
+from fulltext import get, backend_from_fname, backend_from_fobj, is_file_path
+
+
+def _orig_fname(s):
+    s = basename(s)
+    if s.lower().endswith('.gz'):
+        s = s[:-3]
+    return s
+
+
+def _has_ext(s):
+    return bool(splitext(s)[1])
+
+
+def _get_file(path_or_file, **kwargs):
```

Comment:
You may want to provide both `_get_file()` and `_get_path()` because providing just `_get_file()` means that fulltext, when passed a path will always open the file itself and call `_get_file()`. OTOH, exporting `_get_path()` will allow it to just pass the path, and GzipFile will then handle it.

Reply:
> means that fulltext, when passed a path will always open the file itself and call _get_file()

OK. I'll just do:

```
_get_path = _get_file
```

...in `__gz.py` then (`__zip.py` does the same).

Reply:
pushed; not sure if I did what you meant